### ISSUE TITLE
[iOS] Undefined symbols for architecture arm64

### ISSUE THREAD

**Issue by Mol0ko** [devtodev-analytics/Unity-sdk-3.0#6]
I use gitlab-ci to build Unity project for iOS on macOS runner. iOS build job fails since I added DevToDev unity sdk to my project. Error log:

```
Undefined symbols for architecture arm64:
  "_adImpression", referenced from:
      _Analytics_adImpression_mF1EA45150B9B1C854F932985F357BBED87420C8F in AnalyticsIOS1.o
     (maybe you meant: _Analytics_adImpression_mF1EA45150B9B1C854F932985F357BBED87420C8F)
  "_applyConfig", referenced from:
      _PlatformRemoteConfig_applyConfig_mFF51203DADA2290B10060E4E8F21028CB957DBD5 in AnalyticsIOS2.o
      _PlatformRemoteConfig_ApplyConfig_mFE578E50CC9ECED36B97C13C7C78F515CF7BCB1A in AnalyticsIOS2.o
     (maybe you meant: _PlatformRemoteConfig_applyConfig_mFF51203DADA2290B10060E4E8F21028CB957DBD5)
  "_cacheTestExperiment", referenced from:
      _PlatformRemoteConfig_cacheTestExperiment_mD4993E05B5977AECB4F67562D514C5579AB54272 in AnalyticsIOS2.o
      _PlatformRemoteConfig_CacheTestExperiment_m74CEF1DE31EE2ADE2F41CC61A565FBCF3CEFC0C3 in AnalyticsIOS2.o
     (maybe you meant: _PlatformRemoteConfig_cacheTestExperiment_mD4993E05B5977AECB4F67562D514C5579AB54272)
  "_clearUser", referenced from:
      _UserCard_ClearUser_mBB7BC1B89D3AD3ED7141F31477AB72350E2D48A5 in AnalyticsIOS2.o
      _UserCard_clearUser_m107F1CCBEF5DC08D492B350B627E91CA77CC14EA in AnalyticsIOS2.o
     (maybe you meant: _UserCard_clearUser_m107F1CCBEF5DC08D492B350B627E91CA77CC14EA)
  "_coppaControlEnable", referenced from:
      _Analytics_coppaControlEnable_m3CE488D2F5498F71324FAA2E5F829581A40A69A7 in AnalyticsIOS1.o
      _Analytics_CoppaControlEnable_m7682A6D08117B96084E8BA896E85798F916D3D37 in AnalyticsIOS1.o
     (maybe you meant: _Analytics_coppaControlEnable_m3CE488D2F5498F71324FAA2E5F829581A40A69A7)
  "_currencyAccrual", referenced from:
      _Analytics_currencyAccrual_mA1FD748B99ADBA6DE0C1575DF13FC1373638C892 in AnalyticsIOS1.o
     (maybe you meant: _Analytics_currencyAccrual_mA1FD748B99ADBA6DE0C1575DF13FC1373638C892)
  "_currentBalance", referenced from:
      _Analytics_currentBalance_m54AF5EE7E1102F5669E478C5ADFB2CE60F79210D in AnalyticsIOS1.o
     (maybe you meant: _Analytics_currentBalance_m54AF5EE7E1102F5669E478C5ADFB2CE60F79210D)
  "_customEvent", referenced from:
      _Analytics_customEvent_m7A3ED22131DBC9390F708F43B25C8203E105DC35 in AnalyticsIOS1.o
     (maybe you meant: _Analytics_customEvent_m7A3ED22131DBC9390F708F43B25C8203E105DC35)
  "_enableLogger", referenced from:
      _Analytics_enableLogger_m5299EC97485CCE627412239B3B6C1FD07BF530EF in AnalyticsIOS1.o
      _Analytics_OnDebugMessage_m714F39C95C0FBC3A8768091087065282379B711D in AnalyticsIOS1.o
     (maybe you meant: _Analytics_enableLogger_m5299EC97485CCE627412239B3B6C1FD07BF530EF)
  "_finishProgressionEvent", referenced from:
      _Analytics_finishProgressionEvent_m0D62621316872EACCD7C4C4C22822049558FCBC4 in AnalyticsIOS1.o
      _Analytics_FinishProgressionEvent_mE897C62CCC6EA01508AEB06D7F6947B9C20A930C in AnalyticsIOS1.o
     (maybe you meant: _Analytics_finishProgressionEvent_m0D62621316872EACCD7C4C4C22822049558FCBC4, _Analytics_finishProgressionEventWithParameters_mE95DCC37D00C0DDFC17CC28BC70FB259D002AFDE )
  "_finishProgressionEventWithParameters", referenced from:
      _Analytics_finishProgressionEventWithParameters_mE95DCC37D00C0DDFC17CC28BC70FB259D002AFDE in AnalyticsIOS1.o
     (maybe you meant: _Analytics_finishProgressionEventWithParameters_mE95DCC37D00C0DDFC17CC28BC70FB259D002AFDE)
  "_get", referenced from:
  ...
```

Interesting thing is the error throws every time I build iOS for the first time after git clone (with no `/Libraries` folder). Meanwhile second build (when `/Libraries` folder already generated) produces valid iOS project without build errors. So I suppose that DevToDev caches in Libraries generates incorrectly for the first time somehow.

 It is Unity batchmode build from CLI, not from unity instance (with UI).
Here is my bash command to build iOS:
```
${PATH_TO_UNITY} -projectPath ${CI_PROJECT_DIR} -batchmode -quit -logfile ${LOGFILE} -executeMethod CI.CiBuild.BuildIOS ${XCODE_PROJECT}
```
Here is CI.CiBuild.BuildIOS:
```
        public static void BuildIOS()
        {
            const BuildTarget target = BuildTarget.iOS;
            PlayerSettings.SplashScreen.showUnityLogo = false;
            PlayerSettings.iOS.targetDevice = iOSTargetDevice.iPhoneOnly;
            BuildPipeline.BuildPlayer(GetScenes(), Environment.GetCommandLineArgs().Last(), target, BuildOptions.None);
        }
```

Environment:

_Unity 2021.3.8f1
Xcode 14.2
MacOs Ventura 13.0.1
DevToDev SDK 3.4.0 (installed with package manager)_

As a temporary solution (I hope) I build iOS project twice from gitlab job. 
Please help. Thanks!

**Comment by kaikristo** [comment]
fixed in v3.8.1

**Comment by kaikristo** [comment]
fixed in v3.8.0
